instrumentation/restlet/restlet-1.0/library/src/main/java/io/opentelemetry/instrumentation/restlet/v1_0/TracingFilter.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.restlet.v1_0;
+
+import static io.opentelemetry.instrumentation.api.servlet.ServerSpanNaming.Source.CONTROLLER;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.servlet.ServerSpanNaming;
+import io.opentelemetry.instrumentation.restlet.v1_0.internal.RestletServerSpanNaming;
+import org.restlet.Filter;
+import org.restlet.data.Request;
+import org.restlet.data.Response;
+import org.restlet.data.Status;
+
+final class TracingFilter extends Filter {
+
+  private static final ThreadLocal<ContextAndScope> otelContextAndScope = new ThreadLocal<>();
+  private final Instrumenter<Request, Response> instrumenter;
+  private final String path;
+
+  public TracingFilter(Instrumenter<Request, Response> instrumenter, String path) {
+    this.instrumenter = instrumenter;
+    this.path = path;
+  }
+
+  @Override
+  public int doHandle(Request request, Response response) {
+    try {
+      super.doHandle(request, response);
+    } catch (Throwable t) {
+      response.setStatus(new Status(Status.SERVER_ERROR_INTERNAL, t));
```

Comment:
You could implement the whole instrumentation inside `doHandle` to avoid modifying the response object

Reply:
Without setting status this way, the throwable is not propagated and we get a response with status of 200, not 500. My approach based on [StatusFilter](http://repository.transtep.com/repository/thirdparty/restlet/1.1.6/docs/nre/com/noelios/restlet/StatusFilter.html#doHandle(org.restlet.data.Request,%20org.restlet.data.Response)). 
I can move instrumentation code to `doHandle`, leaving this line as it is.

Reply:
I see. How about you move the instrumentation code to `doHandle` and use the `StatusFilter` in library instrumentation tests? We generally want an instrumentation to be as transparent as possible - I agree that returning 200 for unhandled exceptions is a bit weird and unintuitive, but as instrumentation authors we should avoid changing this default behavior on our whim. If a user wants to return 500s instead then they're going to use `StatusFilter` anyway.

Reply:
Sure, thanks for the explanation!